src/ResultWriter/faultoutput.f90

```diff
@@ -535,174 +541,179 @@ SUBROUTINE calc_FaultOutput( DynRup_output, DISC, EQN, MESH, MaterialVal, BND, t
+              cos1 = dot_product(strike_vector(:),NormalVect_s(:))
+              crossprod(:) = strike_vector(:) .x. NormalVect_s(:)
+
+              scalarprod = dot_product(crossprod(:),NormalVect_n(:))
+              !TU 2.11.15 :cos1**2 can be greater than 1 because of rounding errors -> min
+              IF (scalarprod.GT.0) THEN
+                  sin1=sqrt(1-min(1d0,cos1**2))
+              ELSE
+                  sin1=-sqrt(1-min(1d0,cos1**2))
+              ENDIF
 
-              IF (DynRup_output%OutputMask(6).EQ.1) THEN
-                  ! TU 07.15 rotate Slip from face reference coordinate to (strike,dip, normal) reference cordinate
-                  cos1 = dot_product(strike_vector(:),NormalVect_s(:))
-                  crossprod(:) = strike_vector(:) .x. NormalVect_s(:)
+              OutVars = OutVars + 1
+              DynRup_output%OutVal(iOutPoints,1,OutVars)  = cos1 * DISC%DynRup%output_Slip1(iBndGP,iFace) - sin1* DISC%DynRup%output_Slip2(iBndGP,iFace) !OutVars = 12
+              OutVars = OutVars + 1
+              DynRup_output%OutVal(iOutPoints,1,OutVars)  = sin1 * DISC%DynRup%output_Slip1(iBndGP,iFace) + cos1 * DISC%DynRup%output_Slip2(iBndGP,iFace) !OutVars = 13
+          ENDIF
 
-                  scalarprod = dot_product(crossprod(:),NormalVect_n(:))
-                  !TU 2.11.15 :cos1**2 can be greater than 1 because of rounding errors -> min
-                  IF (scalarprod.GT.0) THEN
-                      sin1=sqrt(1-min(1d0,cos1**2))
-                  ELSE
-                      sin1=-sqrt(1-min(1d0,cos1**2))
-                  ENDIF
+          IF (DynRup_output%OutputMask(7).EQ.1) THEN
+            ! Thomas Ulrich 3.08.2015
+            ! Calculation of rupture velocity from rupture time array
+            ! using the 2d basis functions and their derivatives
+
+            ! To compute accurately dt/dxi, all rupture_time have to be defined on the element
+            compute_VR = .TRUE.
+            DO jBndGP = 1,DISC%Galerkin%nBndGP
+               IF (DISC%DynRup%output_rupture_time(jBndGP,iFace).EQ.0) THEN
+                  compute_VR = .FALSE.
+                  Vr = 0d0
+               ENDIF
+            ENDDO
 
-                  OutVars = OutVars + 1
-                  DynRup_output%OutVal(iOutPoints,1,OutVars)  = cos1 * DISC%DynRup%output_Slip1(iBndGP,iFace) - sin1* DISC%DynRup%output_Slip2(iBndGP,iFace)
-                  OutVars = OutVars + 1
-                  DynRup_output%OutVal(iOutPoints,1,OutVars)  = sin1 * DISC%DynRup%output_Slip1(iBndGP,iFace) + cos1 * DISC%DynRup%output_Slip2(iBndGP,iFace)
-              ENDIF
+            IF (compute_VR.EQ..TRUE.) THEN
+
+            ! The rupture velocity is based on the gradient of the rupture time
+            ! The gradient is not expected to be accurate for the side gauss points
+            !! We consider then the next internal gauss point if the closest gauss point is external
```

Comment:
Where do all these changes come from? Please don't "just" change the formatting and indentation.